ioservice/io_foms.c

```diff
@@ -2152,24 +2182,30 @@ static int m0_io_fom_cob_rw_tick(struct m0_fom *fom)
 				m0_cob_tx_credit(fom_cdom(fom),
 						 M0_COB_OP_DELETE, accum);
 			}
-		} else if (m0_fom_phase(fom) == M0_FOPH_AUTHORISATION) {
+		} else if (phase == M0_FOPH_AUTHORISATION) {
 			rc = m0_fom_tick_generic(fom);
 			if (m0_fom_phase(fom) == M0_FOPH_TXN_INIT)
 				m0_fom_phase_set(fom, M0_FOPH_IO_FOM_PREPARE);
 			return M0_RC(rc);
-		} else if (m0_fom_phase(fom) == M0_FOPH_TXN_WAIT) {
+		} else if (phase == M0_FOPH_TXN_WAIT) {
 			rc = m0_fom_tick_generic(fom);
 			if (m0_fom_phase(fom) == M0_FOPH_IO_FOM_PREPARE)
 				m0_fom_phase_set(fom, M0_FOPH_IO_STOB_INIT);
 			return M0_RC(rc);
+		} else if (phase == M0_FOPH_TXN_COMMIT) {
+			rc = m0_fom_tick_generic(fom);
+			M0_ASSERT(rc == M0_FSO_AGAIN);
+			M0_ASSERT(m0_fom_phase(fom) == M0_FOPH_QUEUE_REPLY);
+			if (m0_is_write_fop(fom->fo_fop))
+				m0_fom_phase_set(fom, M0_FOPH_IO_SYNC);
+			return M0_RC(rc);
```

Comment:
@madhavemuri @nikitadanilov , in the commit of object creation sync, @madhavemuri just used the same "M0_FOPH_QUEUE_REPLY" to wait for the TX.
That makes it simpler.
Can we do the same for I/O?
Or we can keep it as of now.

Reply:
For cob_foms it will need some time, this can be done as improvement.
Also once DTM is ready, we may need to re-organize here.

Reply:
OK